ssz-rs/src/de.rs

```diff
@@ -93,26 +102,45 @@ where
         return Ok(vec![])
     }
 
-    let data_pointer = u32::deserialize(&encoding[..BYTES_PER_LENGTH_OFFSET])?;
-    let data_pointer = data_pointer as usize;
-    if encoding.len() < data_pointer {
+    let offsets_len =
+        encoding.get(..BYTES_PER_LENGTH_OFFSET).ok_or(DeserializeError::ExpectedFurtherInput {
+            provided: encoding.len(),
+            expected: BYTES_PER_LENGTH_OFFSET,
+        })?;
+    let offsets_len = u32::deserialize(offsets_len)?;
+    let offsets_len = offsets_len as usize;
+    if encoding.len() < offsets_len {
         return Err(DeserializeError::ExpectedFurtherInput {
             provided: encoding.len(),
-            expected: data_pointer,
+            expected: offsets_len,
         })
     }
+    if offsets_len % BYTES_PER_LENGTH_OFFSET != 0 {
+        return Err(DeserializeError::IncompleteLengthOffsets(offsets_len));
+    }
 
-    let offsets = &mut encoding[..data_pointer]
+    let offsets = &mut encoding
+        .get(..offsets_len)
+        .ok_or(DeserializeError::ExpectedFurtherInput {
+            provided: encoding.len(),
+            expected: offsets_len,
+        })?
```

Comment:
we can drop this as we just verified `encoding.len() >= offsets_len` on line 112